modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -568,23 +617,46 @@ namespace cv {
                             padding = kernel_size / 2;
 
                         CV_Assert(kernel_size > 0 && filters > 0);
-                        CV_Assert(current_channels > 0);
+                        CV_Assert(tensor_shape[0] > 0);
 
-                        setParams.setConvolution(kernel_size, padding, stride, filters, current_channels,
+                        setParams.setConvolution(kernel_size, padding, stride, filters, tensor_shape[0],
                             batch_normalize);
 
-                        current_channels = filters;
+                        tensor_shape[0] = filters;
+                        tensor_shape[1] = (tensor_shape[1] - kernel_size + 2 * padding) / stride + 1;
+                        tensor_shape[2] = (tensor_shape[2] - kernel_size + 2 * padding) / stride + 1;
+                    }
+                    else if (layer_type == "connected")
+                    {
+                        int output = getParam<int>(layer_params, "output", 1);
+                        bool batch_normalize = getParam<int>(layer_params, "batch_normalize", 0) == 1;
+
+                        CV_Assert(output > 0);
+
+                        setParams.setFullyConnected(output, batch_normalize);
+
+                        if(layers_counter && tensor_shape[1] > 1)
```

Comment:
What this `if` condition means?

Reply:
```
if(layers_counter && tensor_shape[1] > 1)
            net->out_channels_vec[layers_counter-1] = total(tensor_shape);
```
If `Connected` is first layer of model - `layers_counter = zero` so wanted to avoid this situation `net->out_channels_vec[ -1 ] = total(tensor_shape);`

`tensor_shape[1] > 1` It makes sure that current connected layer if first connected layer after conv or other layer, just like

```
1. [conv] 
2. [conv]
3. [connected]
4. [connected]
```
so it only applies for `3. [connected] ` 
For `4. [connected]` and later on layers, no need to use `net->out_channels_vec[layers_counter-1] = total(tensor_shape);`. 

For all other layers, this will handle.
https://github.com/opencv/opencv/blob/24932c781ea872cf542536402793b43a8d8c0216/modules/dnn/src/darknet/darknet_io.cpp#L769